### PR TITLE
[DSEW CPR] Pass through python-black

### PR DESCRIPTION
**Pull request by dshemetov**
### Description
Something of a personal preference, as I am usually into [black's output](https://pypi.org/project/black/). Some main changes:
- consistent white spaces
- consistent use of " versus '
- takes advantage of () to separate long pandas lines, i.e.
```py
# Turn this 
pd.merge(df, max_ts_by_group, on=["publish_date", "geo_id"], how="outer").assign(is_max_group_ts=lambda df: df["timestamp"] == df["max_timestamp"]).drop(["max_timestamp"], axis=1))

# Into this
(
  pd.merge(df, max_ts_by_group, on=["publish_date", "geo_id"], how="outer")
  .assign(is_max_group_ts=lambda df: df["timestamp"] == df["max_timestamp"])
  .drop(["max_timestamp"], axis=1)
)
```

Merge after the interpolation work.

### CODE REVIEW

**Comment by dshemetov** [comment]
@krivard let's merge this into #1555 only when we're ready to merge that one, so we don't clutter the diffs there.

**Comment by dshemetov** [comment]
Also let me know if you have any objections to this. It doesn't look great in some parts (like where it has to break many composed functions up and it creates too much white space by creating too many lines with just a single open bracket), but at least it's consistent.

**Review by krivard — COMMENTED** [review]
Okay this is a really detailed pass but I don't want it to be overwhelming so here's a rough guide:

General case:
* quote normalization is universally 💯 
* space-padding around operators, assignment, and parameter specification is universally 💯 
* using parens to reduce the need for backslashes and permit the method name to be the first item on a line is almost always 💯 
* same-line vs multi-line formatting decisions are too context-dependent to be able to rely on an automated tool 👎 

Specifics:
* I marked particularly good reformats with 💯 
* I marked particularly bad reformats with "this is worse"
* I marked borderline reformats with "ehhhhhh"
* Anything unmarked is either covered by one of the universal 💯 s above, or was a moderate improvement, neutral, or escaped notice

I'm not 100% sure what this means -- Black doesn't seem to let you turn off particular formatting rules. There does seem to be a mechanism for identifying blocks of code that Black shouldn't touch, but then it wouldn't do the things we do like, and we'd have to maintain an awful lot of those block markers.

File: dsew_community_profile/tests/test_pull.py [hunk 1]
```diff
@@ -32,151 +39,208 @@ def _assert_frame_equal(df1, df2, index_cols: List[str] = None):
...
+                            f"{i}", f"2021010{i}.xlsx", date(2021, 1, i), f"{i}---2021010{i}.xlsx"
+                        ),
+                        inst(
+                            f"p{i}", f"2021010{i}.pdf", date(2021, 1, i), f"p{i}---2021010{i}.pdf"
+                        ),
+                    ]
+                    for i in [1, 2, 3, 4, 5]
+                ]
+            )
+        )
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:236]: yea this was the worst one, it tried so hard

File: dsew_community_profile/tests/test_pull.py [hunk 2]
```diff
@@ -206,177 +271,174 @@ def test_nation_from_state(self):
...
+        test_df = pd.DataFrame(
+            {
+                "state_id": ["pa", "wv"],
+                "timestamp": [datetime(year=2020, month=1, day=1)] * 2,
+                "val": [15.0, 150.0],
+                "se": [None, None],
+                "sample_size": [None, None],
+                "publish_date": [datetime(year=2020, month=1, day=1)] * 2,
+            }
+        )
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:283]: agree

File: dsew_community_profile/tests/test_pull.py [hunk 3]
```diff
@@ -206,177 +271,174 @@ def test_nation_from_state(self):
...
 
         pd.testing.assert_frame_equal(
-            generate_prop_signal(
-                test_df.copy(),
-                "msa",
-                geomapper
-            ),
-            expected_df,
-            check_like=True
+            generate_prop_signal(test_df.copy(), "msa", geomapper), expected_df, check_like=True
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:352]: i'm more ok with things like this. save vertical space, whole test is one line, and the commas and parens aren't that hard to parse here, imo

File: dsew_community_profile/tests/test_pull.py [hunk 4]
```diff
@@ -386,50 +448,55 @@ def test_unify_testing_sigs(self):
...
             )
 
     def test_std_err(self):
-        df = pd.DataFrame({
-            "val": [0, 0.5, 0.4, 0.3, 0.2, 0.1],
-            "sample_size": [2, 2, 5, 10, 20, 50]
-        })
+        df = pd.DataFrame(
+            {"val": [0, 0.5, 0.4, 0.3, 0.2, 0.1], "sample_size": [2, 2, 5, 10, 20, 50]}
+        )
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:499]: definitely not great

File: dsew_community_profile/tests/test_pull.py [hunk 5]
```diff
@@ -32,151 +39,208 @@ def _assert_frame_equal(df1, df2, index_cols: List[str] = None):
...
+            ),
+            example(
+                "RT-PCR tests per 100k - last 7 days (may be an underestimate due to delayed reporting)",
+                False,
+            ),
+            example("Confirmed COVID-19 admissions - last 7 days", True),
+            example("Confirmed COVID-19 admissions - percent change", False),
+            example("Confirmed COVID-19 admissions - last 7 days - ages <18", False),
+            example("Confirmed COVID-19 admissions - last 7 days - age unknown", False),
+            example("Confirmed COVID-19 admissions per 100 inpatient beds - last 7 days", False),
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:209]: yea, the main thing it's trying to do here is appease the line-length constraints. since to lines vary in length by a lot, there wont be a single setting that will make them all consistent (except for making them all one line, which is also something we don't want)

File: dsew_community_profile/tests/test_pull.py [hunk 6]
```diff
@@ -32,151 +39,208 @@ def _assert_frame_equal(df1, df2, index_cols: List[str] = None):
...
+            ),
+            example("TESTING: % CHANGE FROM PREVIOUS WEEK", True),
+            example("VIRAL (RT-PCR) LAB TESTING: % CHANGE FROM PREVIOUS WEEK", True),
+            example("TESTING: DEMOGRAPHIC DATA", True),
+            example("HOSPITAL UTILIZATION: LAST WEEK (January 2-8)", False),
+            example("HOSPITAL UTILIZATION: CHANGE FROM PREVIOUS WEEK", True),
+            example("HOSPITAL UTILIZATION: DEMOGRAPHIC DATA", True),
+            example("COVID-19 VACCINATION DATA: CUMULATIVE (January 25)", False),
+            example("COVID-19 VACCINATION DATA: LAST WEEK (January 25-31)", False),
+            example("COVID-19 VACCINATION DATA: DEMOGRAPHIC DATA", True),
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:166]: agreed

File: dsew_community_profile/tests/test_pull.py [hunk 7]
```diff
@@ -32,151 +39,208 @@ def _assert_frame_equal(df1, df2, index_cols: List[str] = None):
...
+                DatasetTimes("last", None, None, date(2021, 7, 8), None, None),
+            ),
+            example(
+                "COVID-19 VACCINATION DATA: CUMULATIVE (January 25)",
+                DatasetTimes("", None, None, None, date(2021, 1, 25), None),
+            ),
+            example(
+                "COVID-19 VACCINATION DATA: LAST WEEK (January 25-31)",
+                DatasetTimes("last", None, None, None, None, date(2021, 1, 25)),
+            ),
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/tests/test_pull.py:131]: i think this is one of the rare instances where it got lucky and the single line-length == 100 setting worked for all the examples. seems like it's best for consistent formatting when the lines are all about the same length

File: dsew_community_profile/delphi_dsew_community_profile/run.py [hunk 8]
```diff
@@ -63,13 +64,9 @@ def replace_date_param(p):
         if sig not in params["indicator"]["export_signals"]:
             continue
         dates = create_export_csv(
-            df,
-            params['common']['export_dir'],
-            geo,
-            make_signal_name(sig, is_prop),
-            **export_params
+            df, params["common"]["export_dir"], geo, make_signal_name(sig, is_prop), **export_params
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/delphi_dsew_community_profile/run.py:67]: i think im ok with this. i like that `create_export_csv` doesnt take up 5 lines of vertical space and if i dont care about its arguments, i can just read the function name and move on.

File: dsew_community_profile/delphi_dsew_community_profile/run.py [hunk 9]
```diff
@@ -36,25 +36,26 @@ def run_module(params):
...
-        'start_date': params["indicator"].get("export_start_date", None),
-        'end_date': params["indicator"].get("export_end_date", None)
-    }
-    export_params = {
-        k: pd.to_datetime(v) if v is not None else v
-        for k, v in export_params.items()
+        "start_date": params["indicator"].get("export_start_date", None),
+        "end_date": params["indicator"].get("export_end_date", None),
     }
+    export_params = {k: pd.to_datetime(v) if v is not None else v for k, v in export_params.items()}
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/delphi_dsew_community_profile/run.py:58]: i think im too far gone in python list comprehension territory, since im ok with this

File: dsew_community_profile/delphi_dsew_community_profile/pull.py [hunk 10]
```diff
@@ -167,68 +181,76 @@ def __init__(self, config, sheets=TRANSFORMS.keys(), logger=None):
...
+                    header.find("WEEK (") > 0
+                )
+                or  # include "COVID-19 VACCINATION DATA: CUMULATIVE (January 25)"
+                # include "COVID-19 VACCINATION DATA: LAST WEEK (January 25-31)"
+                (
+                    header.startswith("COVID-19 VACCINATION DATA: CUMULATIVE")
+                    or header.startswith("COVID-19 VACCINATION DATA: LAST WEEK")
+                )
+            )
+        )
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/delphi_dsew_community_profile/pull.py:244]: agreed, that's probably the right solution here

File: dsew_community_profile/delphi_dsew_community_profile/pull.py [hunk 11]
```diff
@@ -515,24 +558,22 @@ def fetch_new_reports(params, logger=None):
...
-                    drop=True
-                ) == 1), f"Duplicate rows in {sig} indicate that one or" \
-                + " more reports were published multiple times and the copies differ"
+            assert all(
+                latest_sig_df.groupby(["timestamp", "geo_id"]).size().reset_index(drop=True) == 1
+            ), (
+                "Duplicate"
+                + f" rows in {sig} indicate that one or more reports were"
+                + " published multiple times and the copies differ."
+            )
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/delphi_dsew_community_profile/pull.py:576]: yea i'm not a fan of two parens-broken statements with a comma in between

File: dsew_community_profile/delphi_dsew_community_profile/pull.py [hunk 12]
```diff
@@ -586,20 +624,42 @@ def interpolate_missing_values(dfs: DataDict) -> DataDict:
...
-                reindexed_group_df["sample_size"] = reindexed_group_df["sample_size"].interpolate(method="linear", limit_area="inside").astype(float)
+                reindexed_group_df["se"] = (
+                    reindexed_group_df["se"]
+                    .interpolate(method="linear", limit_area="inside")
+                    .astype(float)
+                )
+            if (
+                "sample_size" in reindexed_group_df.columns
+                and not reindexed_group_df["sample_size"].isna().all()
+            ):
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply dsew_community_profile/delphi_dsew_community_profile/pull.py:646]: lol "i split this line, but im not happy about it" 

it's the line-length thing again!

**Comment by dshemetov** [comment]
Hm yea, well it's good to learn that same line versus multi-line decisions are really a big weakness of this tool. 

Maybe we just keep this a one-off run, keep the parts we like, reject parts we don't, and move on?

**Review by krivard — COMMENTED** [review]
should we refresh this or trash it?

**Comment by dshemetov** [comment]
I could go either way:
- trash: inconsistency is tolerable, fixing it is low priority, also, it's a bit annoying to have to write `#fmt: off` around blocks where we don't like the formatting
- refresh: consistency is nice, it's nice to have it automated, so we can focus developer time on more important things

In my personal projects, I use black and just accept its formatting, because I prefer to just not think about it. But as for using it on our team, maybe we should poll our developers' preferences instead.

**Comment by nmdefries** [comment]
We can close this, [CPR has been deprecated](https://github.com/cmu-delphi/covidcast-indicators/pull/1871).
